resteasy-jaxrs/src/main/java/org/jboss/resteasy/core/ServerResponseWriter.java

```diff
@@ -78,7 +78,8 @@ public static void writeNomapResponse(BuiltResponse jaxrsResponse, final HttpReq
       
       executeFilters(jaxrsResponse, request, response, providerFactory, method);
 
-      if (jaxrsResponse.getEntity() == null)
+      //[RESTEASY-1627] check on response.getOutputStream() to avoid resteasy-netty4 trying building a chunked response body for HEAD requests 
+      if (jaxrsResponse.getEntity() == null || response.getOutputStream() == null)

```

Comment:
I've been debugging and thinking a bit about this. We can't remove headers from HttpServletResponse instances, so an attempt of removing the Content-Length header from the response would fail. This said, when commitHeader is invoked, the Content-Length header is not set yet in the case of resteasy-undertow and wildfly, I assume because Undertow sets it later on. So I decided to at least complete the proposed fix here, by covering all the adapters & providing tests; this means either preventing the Content-Length header from being emitted if that's something we can control (resteasy-netty4, resteasy-vertx, resteasy-jdk-http) or by having that computed properly instead of being set to 0 (resteasy-netty, resteasy-undertow, wildfly/eap).
My proposal would be to go with this solution for now, given it basically keeps the current resteasy-netty/resteasy-undertow/resteasy-jdk-http/wildfly/eap behaviors while fixing broken head responses in resteasy-netty4 and resteasy-vertx. We can possibly create a new jira for further investigating unifying all server adapters' behavior when processing HEAD requests.